### PR TITLE
Feature add row

### PR DESCRIPTION
**Pull request by jdmcd**
@EricKuck ready for code review

### CODE REVIEW

File: Wave/StringSetting.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public var key: String
+    
+    public init(placeholder: String?, defaultValue: String?, key: String, value: String, title: String?) {
+        self.placeholder = placeholder
+        self.defaultValue = defaultValue
+        self.key = key
+        self.value = value
+        
+        super.init()
+        self.title = title
```
Comment (EricKuck) [inline Wave/StringSetting.swift:24]: Why's this after the init, while the others are all before?

File: Wave/StringSetting.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        super.configureCell(tableViewCell)
+        
+        guard let currentText = tableViewCell.textLabel?.text else {
+            tableViewCell.textLabel?.text = value
+            return
+        }
+        tableViewCell.textLabel?.text = "\(currentText) \(value)"
+    }
+    
+    override public func store() {
```
Comment (EricKuck) [inline Wave/StringSetting.swift:33]: Let's add an implementation to this.

File: Wave/StringSetting.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        self.value = value
+        
+        super.init()
+        self.title = title
+    }
+    
+    public override func configureCell(tableViewCell: UITableViewCell) {
+        super.configureCell(tableViewCell)
+        
+        guard let currentText = tableViewCell.textLabel?.text else {
```
Comment (EricKuck) [inline Wave/StringSetting.swift]: I'm a little confused about what this is doing. What's the goal here?

File: Wave/Wave/SingleSelectionSetting.swift [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        self.possibleValues = possibleValues
+        self.enableCustom = enableCustom
+        self.customOption = customOption
+        self.name = name
+        
+        for value in self.possibleValues {
+            self.settings.append(StringSetting(placeholder: nil, defaultValue: nil, key: "\(name)_\(value)", value: String(value), title: nil))
+        }
+        
+        if self.enableCustom {
```
Comment (EricKuck) [inline Wave/Wave/SingleSelectionSetting.swift]: Booo `self`

File: Wave/Wave/WaveKeeper.swift [hunk 5]
```diff
@@ -9,5 +9,5 @@
 import UIKit
 
 public final class WaveKeeper {
-    //The class that will hold all of the store data, and manage its' persistance 
+    public static let sharkedInstance = WaveKeeper()
```
Comment (EricKuck) [inline Wave/Wave/WaveKeeper.swift]: I mean I guess it IS shark week, but this still seems like a weird choice.

Comment (EricKuck) [inline Wave/Wave/WaveKeeper.swift]: Also if you're going to create a singleton, make the initializer private.

File: Wave/Wave/WaveViewController.swift [hunk 6]
```diff
@@ -22,8 +22,16 @@ final public class WaveViewController: UIViewController {
         return tableView
     }()
     
+    func dismiss() {
+        dismissViewControllerAnimated(true, completion: nil)
+    }
+    
     override public func loadView() {
         super.loadView()
+        let doneButton = UIBarButtonItem(barButtonSystemItem: .Done, target: self, action: #selector(dismiss))
```
Comment (EricKuck) [inline Wave/Wave/WaveViewController.swift]: Why store this in a let if all you're doing with it is making it the right bar button?

File: Wave/WaveExample/AppDelegate.swift [hunk 7]
```diff
@@ -17,7 +17,20 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(application: UIApplication, didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
         // Override point for customization after application launch.
         
-        window = WaveWindow(frame: UIScreen.mainScreen().bounds, style: .RightDrawer, sections: [Section()], enabled: true)
+        let stringSetting = StringSetting(placeholder: "", defaultValue: nil, key: "mainEndpoint", value: "http://www.healthipass.com/api/v2/", title: "Main Endpoint:")
```
Comment (EricKuck) [inline Wave/WaveExample/AppDelegate.swift]: Let's make this more generic for an open source project.

File: Wave/Wave/SingleSelectionSetting.swift (hunk 4 above)
Comment (jdmcd) [inline Wave/Wave/SingleSelectionSetting.swift]: 👻 👻

File: Wave/StringSetting.swift (hunk 2 above)
Comment (EricKuck) [inline Wave/StringSetting.swift:33]: Just kidding! Not sure went I wrote that, it should def be another PR.

Comment (jdmcd) [inline Wave/StringSetting.swift:33]: Yeah I was gonna ask about that lol

File: Wave/StringSetting.swift (hunk 1 above)
Comment (jdmcd) [inline Wave/StringSetting.swift:24]: Because `title` comes from `Setting` and if it's moved before the `super.init()` then it complains that self hasn't been initialized yet

File: Wave/WaveExample/AppDelegate.swift (hunk 7 above)
Comment (jdmcd) [inline Wave/WaveExample/AppDelegate.swift]: oops sorry HP

File: Wave/StringSetting.swift (hunk 3 above)
Comment (jdmcd) [inline Wave/StringSetting.swift]: On second thought this doesn't really make sense

**Comment by jdmcd** [comment]
@EricKuck Good to merge?

File: README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+ ## Wave
```
Comment (EricKuck) [inline README.md:1]: A++ readme. Would read again.

Comment (jdmcd) [inline README.md:1]: I'm pretty good at them

File: Wave/Wave/SingleSelectionSetting.swift [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+public final class SingleSelectionSection: SectionType, SettingType {
+    
+    public var possibleValues: [Any]
+    public var enableCustom = false
+    public var customOption: RowType?
+    public var name: String
+    private var settings = [StringSetting]()
+    
+    public init(possibleValues: [Any], enableCustom: Bool, customOption: RowType?, name: String) {
```
Comment (EricKuck) [inline Wave/Wave/SingleSelectionSetting.swift]: How come this is `[Any]` instead of `[String]`? I guess it could work, as long as whatever is being passed in can be represented as a string. If this was your train of thought, it should probably still be `AnyObject` so it can be serialized to a plist.

Comment (jdmcd) [inline Wave/Wave/SingleSelectionSetting.swift]: That's a good point. I guess automatically converting it to a `StringSetting` is unfair to the developer. Should I switch on the type (right now only Int and String) and then append the setting accordingly? `settings` would have to be of type `[Setting]` which is annoying

Comment (jdmcd) [inline Wave/Wave/SingleSelectionSetting.swift]: My only concern is that if a developer wanted to have a single selection for their port number, for example, we've just converted that port to a string and lost any kind of type integrity

Comment (EricKuck) [inline Wave/Wave/SingleSelectionSetting.swift]: Yeah, I'm not sure what the best way to approach this is. Ideally we would have had this be a generic class, but we all know how that turned out. Second best option might just be to have `StringSingleSelectionSection` and `IntSingleSelectionSection`?

Comment (jdmcd) [inline Wave/Wave/SingleSelectionSetting.swift]: Yeah that's probably the best way to go for right now. I'll do that really quickly, resubmit and then this PR should be good to go

**Comment by jdmcd** [comment]
@EricKuck ready for merge
